Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageImageGallery.swift

```diff
@@ -5,116 +5,81 @@
-    open var interItemSpacing: CGFloat = 2
-
+    /// Content the image gallery should display.
     public var content: [ChatMessageImageAttachment] = [] {
         didSet { updateContentIfNeeded() }
     }
+    
+    override open var intrinsicContentSize: CGSize { .init(width: .max, height: .max) }
 
+    /// Triggered when an attachment is tapped.
     public var didTapOnAttachment: ((ChatMessageImageAttachment) -> Void)?
 
-    // MARK: - Subviews
-
+    /// Previews for images.
     public private(set) lazy var previews = [
         createImagePreview(),
         createImagePreview(),
         createImagePreview(),
         createImagePreview()
     ]
 
-    public private(set) lazy var moreImagesOverlay: UILabel = {
-        let label = UILabel()
-        label.font = appearance.fonts.title
-        label.adjustsFontForContentSizeCategory = true
-        label.textAlignment = .center
-        return label
-            .withoutAutoresizingMaskConstraints
-            .withBidirectionalLanguagesSupport
-    }()
-
-    private var layouts: [[NSLayoutConstraint]] = []
+    /// Overlay to be displayed when `content` contains more images than the gallery can display.
+    public private(set) lazy var moreImagesOverlay = UILabel()
+        .withoutAutoresizingMaskConstraints
+    
+    /// Container holding all previews.
+    public private(set) lazy var previewsContainerView = ContainerStackView()
+        .withoutAutoresizingMaskConstraints
+    
+    /// Left container for previews.
+    public private(set) lazy var leftPreviewsContainerView = ContainerStackView()
+    
+    /// Right container for previews.
+    public private(set) lazy var rightPreviewsContainerView = ContainerStackView()
 
     // MARK: - Overrides
 
     override open func setUpLayout() {
-        previews.forEach(addSubview)
+        previewsContainerView.axis = .horizontal
+        previewsContainerView.distribution = .equal
+        previewsContainerView.alignment = .fill
+        previewsContainerView.spacing = 0
+        embed(previewsContainerView)
+        
+        leftPreviewsContainerView.spacing = 0
+        leftPreviewsContainerView.axis = .vertical
+        leftPreviewsContainerView.distribution = .equal
+        leftPreviewsContainerView.alignment = .fill
+        previewsContainerView.addArrangedSubview(leftPreviewsContainerView)
+        
+        leftPreviewsContainerView.addArrangedSubview(previews[0])
+        leftPreviewsContainerView.addArrangedSubview(previews[2])
+        
+        rightPreviewsContainerView.spacing = 0
+        rightPreviewsContainerView.axis = .vertical
+        rightPreviewsContainerView.distribution = .equal
+        rightPreviewsContainerView.alignment = .fill
+        previewsContainerView.addArrangedSubview(rightPreviewsContainerView)
+        
+        rightPreviewsContainerView.addArrangedSubview(previews[1])
+        rightPreviewsContainerView.addArrangedSubview(previews[3])
```

Comment:
nit: some fancy ASCII art explaining what are the indices of the views within the grid would be cool